### PR TITLE
Bugs fix

### PR DESCRIPTION
**Pull request by lazycipher**
Fix various issues/bugs reported on this repo.
#650 #659 #637 #635 #614 #614 #552 #660

### CODE REVIEW

File: src/utils/allowCredentialsInHeader.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
```
Comment (Rupeshiya) [inline src/utils/allowCredentialsInHeader.js:2]: @lazycipher now you are not using authorization header ?
Only sending cookie?

File: src/App.js [hunk 2]
```diff
@@ -4,15 +4,17 @@ import "./App.css";
 import { Provider } from "react-redux";
 import store from "./store";
 import jwt_decode from "jwt-decode";
-import { setAuthToken } from "./utils/setAuthToken";
```
Comment (Rupeshiya) [inline src/App.js:7]: if authorization is not required in header, remove the function too instead of just removing from here

File: src/App.js [hunk 3]
```diff
@@ -4,15 +4,17 @@ import "./App.css";
...
 import { setCurrentUser, logoutUser } from "./actions/authAction";
 import "./css/main.scss";
 import ReactGA from "react-ga";
 
+allowCredentialsInHeader()
+
 function App() {
   useEffect(() => {
     ReactGA.initialize("UA-173245995-1");
-    setAuthToken()
```
Comment (Rupeshiya) [inline src/App.js:15]: This might effect Analytics part cc @AuraOfDivinity

File: src/actions/adminAction.js [hunk 4]
```diff
@@ -27,31 +26,27 @@ export const createAdmin = (adminInfo) => async (dispatch) => {
...
-     const token = res.data.token;
-     dispatch(setRequestStatus(true));
+      // update state with user
+      localStorage.setItem('userId', res.data.user)
+      dispatch(setCurrentUser(res.data.user._id));
 
-     localStorage.setItem("jwtToken", (token));
-     setAuthToken(token);
+      // update localStorage with admin status 
+      localStorage.setItem('username', `${res.data.user.name.firstName} ${res.data.user.name.lastName}`)
```
Comment (Rupeshiya) [inline src/actions/adminAction.js:38]: Local storage is not that secure!!
Try to come up with some secure approach!!
As of now, you can go with it but it's not acceptable at the production level

**Review by Rupeshiya — CHANGES_REQUESTED** [review]

File: src/actions/adminAction.js (hunk 4 above)
Comment (lazycipher) [reply src/actions/adminAction.js:38]: Setting names in local storage won't do any harm as the protected resources sent from the server need that token in cookie.

File: src/App.js (hunk 3 above)
Comment (lazycipher) [reply src/App.js:15]: It was already there!

Comment (lazycipher) [reply src/App.js:15]: Should I make it work as componentDidMount?

File: src/utils/allowCredentialsInHeader.js (hunk 1 above)
Comment (lazycipher) [reply src/utils/allowCredentialsInHeader.js:2]: Yes after shifting the auth method to use cookies, we're using auth in the cookie header.

File: src/App.js (hunk 2 above)
Comment (lazycipher) [reply src/App.js:7]: I did not understand.

**Comment by vaibhavdaren** [comment]
@Rupeshiya please

**Pull request merged**